backend/tests/end_to_end/infrastructures/apps/load/test_admin_panel.py

```diff
@@ -0,0 +1,231 @@
+        form.is_valid()
+        assert not form.errors
+    change = False
+
+    # When
+    data_admin_panel.save_model(
+        request=HttpRequest(), obj=data, form=form, change=change
+    )
+
+    # Then
+    response = client.get(
+        get_url(
+            path_name="load-detail",
+            path_params={"pk": data.id},
+        )
+    )
+    assert response.status_code == HTTPStatus.OK
+    assert response.data["full_name"] == data.data["full_name"]
+
+
+def test_editing_data(
+    data_admin_panel: admin.DataAdmin,
+):
+    # Given
+    data: Data = model_factories.DataFactory.create()
+
+    new_full_name = "New Full_Name"
+    form = admin.DataAdminForm(
+        data={
+            "data": {
+                "full_name": new_full_name,
+                "age": data.data["age"],
+                "is_satisfied": data.data["is_satisfied"],
+            },
+        },
+        instance=data,
+    )
+
+    with transaction.atomic():
+        form.is_valid()
+        assert not form.errors
+    change = True
+
+    # When
+    data_admin_panel.save_model(
+        request=HttpRequest(), obj=data, form=form, change=change
+    )
+
+    # Then
+    data.refresh_from_db()
+    assert data.data["full_name"] == new_full_name
+
+
+def test_delete_data(
+    unauthenticated_client: APIClientData,
+    data_admin_panel: admin.DataAdmin,
+):
+    # Given
+    client = unauthenticated_client.client
+
+    data: Data = model_factories.DataFactory.create()
+
+    # When
+    data_admin_panel.delete_model(
+        request=HttpRequest(),
+        obj=data,
+    )
+
+    # Then
+    response = client.get(
+        get_url(
+            path_name="load-detail",
+            path_params={"pk": data.id},
+        )
+    )
+    assert response.status_code == HTTPStatus.NOT_FOUND
+
+
+def test_admin_form_raises_validation_error_when_data_is_not_dict():
+    # Given
+    form_data = {"data": "not a dict"}
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_age_is_not_a_number():
+    # Given
+    form_data = {
+        "data": {"full_name": "John Doe", "age": "eleven", "is_satisfied": True}
+    }
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_name_is_not_a_string():
+    # Given
+    form_data = {"data": {"full_name": 12345, "age": 11, "is_satisfied": True}}
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_name_is_satisfied_is_not_bool():
+    # Given
+    form_data = {"data": {"full_name": "John Doe", "age": 11, "is_satisfied": "True"}}
```

Comment:
Is "data" a key work? Is yes, please add it in tests/end_to_end/cosnts and use it in tests.